### DISCUSSION TITLE
Klassische Nulleinspeisung

### DISCUSSION THREAD

**Discussion by NikGro** [hoylabs/OpenDTU-OnBattery#609] [Q&A] [answered]
Frage(n) zum Thema Nulleinspeisung:

Ich sehe in der Dokumentation viele Referenzen zur "Strategie zur Batterieentleerung" die offensichtlich zwei Optionen bietet, die dabei helfen die Anzahl an Zyklen, welche die Batterie durchleben darf, etwas zu reduzieren.

Unterstützt das Projekt auch die ganze klassische Nulleinspeisung? Also dass der Akku durch PV-Überschuss, der über den Hausverbrauch hinaus geht, geladen wird und dass, wenn der Verbrauch höher ist als der aktuelle PV-Output, der Akku, unabhängig von SOC und Tageszeit, Unterstützt und entlädt um zusammen mit der PV auf die benötigte Leistung zu kommen? (Natürlich mit einer festen Untergrenze!)

> Da ich meine eigenen Erklärungen selten verstehe Versuche ich nochmal ein Beispiel anzuhängen:
> 
> Mein Akku soll nie unter 10% kommen und zugleich nie über 90%, unabhängig vom Szenario. 
> 
> Wenn die PV tagsüber 1200W produziert und ich Zuhause nur 600W verbrauche, möchte ich, dass die 600W Verbrauch durch die PV gedeckelt werden und mit den übrigen 600W die Batterie geladen wird (die z.B. gerade bei 20% ist). 
> 
> Jetzt springt aber mein Hausverbrauch plötzlich auf 1800W weil ich das Waffeleisen anschmeiße. Dann möchte ich, dass die 1200W, die von der PV kommen, ins Haus fließen und die fehlenden 600W aus der Batterie ergänzt werden, auch wenn sie mit 20% Recht mager geladen ist. 
> 
> Solange sie nicht unter 10% fällt, ist es mir also egal, ob sie gerade am laden war oder nicht. Und auch wie voll sie geladen ist soll nicht berücksichtigt werden (ausgenommen der festen Ober- & Untergrenze) Ist der entsprechende hohe Verbrauch da, möchte ich die Batterie mitbenutzen, um ihn zu decken. Sinkt der Verbrauch wieder, soll das Aufladen direkt weitergehen. Auch wenn dadurch zwangsweise ein häufigerer Wechsel zwischen "laden" und "entladen" stattfindet und die Lebensdauer der Batterie ggf. leidet.

Vielleicht ist diese Funktion ja auch ganz offensichtlich implementiert und ich übersehe es einfach mangels Möglichkeit das ganze in der Praxis zu testen. (Anschaffung der Anlage läuft noch) Auf mich wirkt es so, dass wenn ich den SOC-Start und SOC-Stopp wert beide auf z.B. 10% setzen würde, vielleicht so halbwegs das erreicht werden könnte woran ich denke.

Und ich Frage mich ob ich den 'Full-Solar-Passthrough' deaktivieren kann (ggf. auf 0% setzen?). Idealerweise würde ich, auch bei vollem Akku, nicht einspeisen wollen, sofern ich keinen Verbrauch habe. Also die "Nulleinspeisung" trotz vollem Akku halten und den Überschuss lieber in wären umwandeln lassen (Ich nehme Mal an, dass das passiert wenn man sowohl Ladegerät als auch Wechselrichter anhält) als ihm dem Netzbetreiber zu schenken.

**Comment by bf4ed** [comment]
Klassische Nulleinspeisung geht. Ohne Batterie oder mit. Ich hab es mit 2 WR gelöst.
Schau hier https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/515#discussion-5762013

**Reply by NikGro** [reply]
Krasses Setup! Ist das gewünschte Ergebnis auch mit nur 1x ESP, 1x HMS1600 und 1x MPPT möglich? Ich bin zwar schwer beeindruckt von dem Umfang deines Setups, steige aber ehrlich gesagt nicht ganz durch bzw würde das ganze gerne so simpel wie möglich halten.

**Reply by bf4ed** [reply]
Für die Nulleinspeisung müssen alle Eingänge von der gleichen Quelle besetzt sein. Das heißt alle mit PV oder alle von der Batterie. Nur so nimmt der WR dann die eingestellte % des gewünschten Ausganges.

Der Teilt das auf. Du kannst nicht von 600W WR mit nur 1 von 2 Eingängen angeschlossen 600W bekommen. Du bekommst nur 300W bei 100%

**Reply by NikGro** [reply]
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/1x-150-35-MPPT-%7C-6x-405Wp-%7C-Pylontech-4,8kWh-48V-%7C-HM%E2%80%90600

Dieses Setup baue ich nach. Also alle Eingänge mit beidem besetzt und dann Weg des geringsten Wiederstands.

Mein eigentliches Verständnisproblem liegt eher in den Einstellungsmöglichkeiten von OpenDTU on Battery. Bzw wie ich das Ganze einstellen muss für das beschriebene Ergebnis. Bestes Beispiel wäre die Art und Weise wie der Zendure PV Hub arbeitet. Auch mit nur einem WR

**Reply by bf4ed** [reply]
Ich hab keinen Victron, deswegen kann ich dir nichts genaues sagen. Ich lade die Batterie mit dem AC Ladegerät von Huawei.

Aber einfach aufbauen und damit spielen. Dann siehst du was du noch brauchst oder wie was zu einstellen ist. Die Nulleinspeisung Einstellungen siehst du bei meinem Aufbau.

**Reply by NikGro** [reply]
Deine Konfiguration habe ich mir angesehen, allerdings erschließt es sich mir leider nicht viel mehr als vorher. Was passiert bei dir zwischen 20% SOC und 35% SOC? Kann die Batterie zwischen diesen beiden Werten nicht einspeisen sondern  NUR laden? Lässt sich das umgehen indem man beide werte identisch setzt bzw. vielleicht auf 10% & 11%?

Und da du Passthrough aus hast, was passiert dann bei dir während die Sonne scheint und das Waffeleisen läuft? Füttert die PV zuerst das Waffeleisen und mit dem Überschuss den Akku? Oder geht ohne die Passthrough-Option alles durch die Batterie? Was passiert wenn die Batterie voll ist, der Verbrauch bei 0 und die Produktion allerdings bei 500w? Werden die 500w in Wärme umgewandelt?

Natürlich hängt das alles ein wenig vom Setup ab. In meinem Fall dürfte der PV-Strom ja eigentlich immer zuerst in den Wechselrichter gehen, geringerer Wiederstand usw. Aber was passiert bei mir dann wenn der Akku voll ist und der Wechselrichter keinen Bedarf hat? Wandelt der MPPT den Rest in Wärme um? Oder wird dann doch eingespeist weil das System nicht weiß wohin mit dem Strom?


Ich warte bis das Setup steht, experimentiere mit den Einstellungen und schaue Mal wie weit ich komme. Vielleicht kooperieren die Bezeichnungen der Einstellungen auch einfach nicht mit meiner Denkweise, wer weiß.

**Reply by spcqike** [reply, marked as answer]
> Wandelt der MPPT den Rest in Wärme um?

ja der MPPT lässt dann die Heizung in sich zum Vorschein kommen ;) 

Nein. Der MPPT wandelt nichts extra in Wärme um. Wenn es keinen Bedarf bzw. Verbraucher gibt (Batterie voll, der wechselrichter speist auch nicht mehr ein), reduziert auch der MPPT die Leistungsaufnahme von den Paneelen. Der Strom „verpufft“ einfach ungenutzt. Wenn der MPPT gar nichts zieht, ist es für die Paneele so, als wären sie gar nicht erst angeschlossen. 

Jeder Verbraucher zieht von der Quelle nur das, was er eben aktuell braucht. (Deine Fernbedienung wird ja auch keine Heizung und wandelt den Strom aus der Batterie, den sie grade nicht zum schalten und senden braucht, in Wärme um, oder?) 

> Ich warte bis das Setup steht, experimentiere mit den Einstellungen und schaue Mal wie weit ich komme. 

genau so. Einfach probieren und gucken was passiert. Dabei klärt sich bestimmt schon vieles auf 😄

**Comment by wuscheltom** [comment]
Hallo Leute,
wenn ich eine Nulleinspeisung auf der DTU installieren möchte, reicht eine .json hozulagen oder muss ich die gesamte ESP neu flashen?

**Reply by spcqike** [reply]
Welche Firmware läuft denn aktuell auf deinem esp?

wenn ein aktuelles openDTU onBattery läuft, reicht es, wenn du es im Dynamic power limiter konfigurierst. Ein konfigurierter powermeter vorausgesetzt.
